docs/developer/other/API.md

```diff
@@ -15,3 +15,94 @@ org.apereo.portal.rest.swagger.SwaggerConfiguration.enabled=true
+   templates
+ * `widgetTemplate`: (may be null) indicates a custom template for rendering the widget, used for
+   the `custom` `widgetType`.
+ * `widgetConfig`: (may be null) JSON configuration that informs the rendering of the widget type or
+    custom widget template. This configuration is shared across all usages of the widget in a given 
+    portlet definition.
+ * `widgetURL` : (may be null) URL from which the browser should read additional JSON to inform the 
+   rendering of the widget. Typically this JSON drives the dynamic portions of the custom widget 
+   template or the widget type rendering.
+ * `staticContent` : (may be null) the static HTML content associated with a simple Simple Content
+   Portlet usage, suitable for direct client-side presentation to the user when attempting to render
+   this portlet.
+ * `pithyStaticContent` : (may be null) IN PRACTICE UNUSED.
+ * `renderOnWeb` : (boolean) when true indicates the portlet is suitable for special rendering in
+   `EXCLUSIVE` window state and then direct rendering the resulting markup by injecting it into the 
+   DOM client-side rather than relying upon the full traditional XSLT rendering pipeline.
+
+Data elements not understood offhand by the author of this documentation:
+
+ * `nodeId` : (This is an opportunity to improve this documentation.)
+ * `url` : ditto.
+
+Example response:
+
+```json
+{
+  "portlet": {
+    "nodeId": "-1",
+    "title": "Welcome to uPortal",
+    "description": "Description of uPortal.",
+    "url": null,
+    "iconUrl": "/ResourceServingWebapp/rs/tango/0.8.90/32x32/mimetypes/text-html.png",
+    "faIcon": null,
+    "fname": "what-is-uportal",
+    "target": null,
+    "widgetURL": null,
+    "widgetType": null,
+    "widgetTemplate": null,
+    "widgetConfig": null,
+    "staticContent": "\n            \n                <h2>What is uPortal?</h2>\n                \n                <p>\n                    <a href=\"http://www.apereo.org/uportal\" target=\"_blank\">uPortal</a>\n                    is a free and open source Java-implemented web portal \n                    platform developed and maintained by participants drawn \n                    from across higher education under the coordination of \n                    <a href=\"http://www.apereo.org/\" target=\"_blank\">Apereo</a>.\n                    uPortal can aggregate content, present self-service \n                    applications, personalize presentation and content on the \n                    basis of groups and user attributes, drive mobile device applications, and allow advanced\n                    end-user-participatory customization of the portal experience. \n                    uPortal supports the JSR-286 and JSR-168 Java portlet specification for\n                    including your custom applications within the portal.\n                </p>\n                \n                <p>Welcome to uPortal.</p> \n            \n        ",
```

Comment:
No objection to this, but in the future we may want to look into using vdom json.
E.G. https://github.com/nolanlawson/vdom-as-json